auth-foundation/src/main/java/com/okta/authfoundation/credential/Credential.kt

```diff
@@ -137,14 +215,29 @@ class Credential internal constructor(
 
     /**
      * The tags associated with this [Credential].
-     *
-     * This can be used when calling [Credential.storeToken] to associate this [Credential] with data relevant to your application.
      */
     val tags: Map<String, String>
         get() {
             return Collections.unmodifiableMap(state.value.tags)
         }
 
+    /**
+     * Set tags for this [Credential].
+     */
+    suspend fun setTags(tags: Map<String, String>) {
+        token?.let {
```

Comment:
Why is token optional here? Is it possible for a Credential's token value to be null?

Reply:
Credential's token value can be null after it has been deleted. In that case, if the user tries setTags, this method throws. I just checked the swift SDK, and I see that it doesn't delete the token stored in the Credential object, but instead removes it from storage and CredentialDataSource. Should I do something similar here? That makes things simpler to maintain on my end.